File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionTestTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionTestTest extends BaseTest {
    @Test
    void testInlineFunctionTest() {
        var input = "hello";
        assertTransformation(input, "$$test([le]):$", true);
        assertTransformation(input, "$$test(hell):$", true);
        assertTransformation(input, "$$test(hello$):$", true);
        assertTransformation(input, "$$test(hell$):$", false);
        assertTransformation(input, "$$test('^hello$'):$", true);
        assertTransformation(input, "$$test('^(hello|world)$'):$", true);
        assertTransformation("HELLO", "$$test('^hello$'):$", false);
        assertTransformation("HELLO", "$$test('(?i)^hello$'):$", true);
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionTest.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionTest", () => {
  test("inline", async () => {
    const input = "hello";
    await assertTransformation(input, "$$test([le]):$", true);
    await assertTransformation(input, "$$test(hell):$", true);
    await assertTransformation(input, "$$test(hello$):$", true);
    await assertTransformation(input, "$$test(hell$):$", false);
    await assertTransformation(input, "$$test('^hello$'):$", true);
    await assertTransformation(input, "$$test('^(hello|world)$'):$", true);
    await assertTransformation("HELLO", "$$test('^hello$'):$", false);
    await assertTransformation("HELLO", "$$test('(?i)^hello$'):$", true);
  });
});
